Dev

smilerz opened the pull request.

smilerz committed:
fixed default value of argparse --sync (#12)

smilerz committed:
added config parser

smilerz committed:
updated README.md to reflect config.ini addition

smilerz committed:
updated help for MediaPlayers

smilerz committed:
added ability to connect to plex with API token

smilerz committed:
updated README.md to include token argument

smilerz committed:
added ability to connect to Plex with token

smilerz committed:
updated MediaMonkey.search_tracks behavior to be consistent with Plex

patzm commented on MediaPlayer.py:
```diff
@@ -158,9 +164,30 @@ def read_track_metadata(self, track):
+		rating = kwargs.get('rating')
+		query = kwargs.get('query')
+		if title: 
+			query = "SongTitle like \"%"+ title +"%\""
```
I think something you might like are f-strings, e.g. 
```suggestion
			query = f'SongTitle like "%{title}%"'
```
also if you want to use `"`, you can use `'` to wrap the string

smilerz commented:
- changed argparse to configargparse
- added config.ini file to minimize required command line arguments
- updated README and help files to better reflect recent changes
- added ability to connect plex with token instead of password

smilerz committed:
Update MediaPlayer.py

Co-authored-by: Martin Patz <5219726+patzm@users.noreply.github.com>

smilerz committed:
Update MediaPlayer.py

patzm commented on requirements.txt:
```diff
@@ -2,3 +2,4 @@ PlexAPI>=4.2.0
 fuzzywuzzy
 pypiwin32
 numpy
+configargparse
```
you should add `ConfigArgParse` to the requirements file.
```suggestion
ConfigArgParse
```
https://pypi.org/project/ConfigArgParse/ this is the package you are using right?

smilerz replied:
yes, that's the one

patzm reviewed (changes requested):
Cool idea!

smilerz committed:
Update requirements.txt

Co-authored-by: Martin Patz <5219726+patzm@users.noreply.github.com>

patzm reviewed (approved):
ok cool :+1: LGTM

PS: you can re-request a review by clicking this double arrow here in the upper right in the PR overview
![image](https://user-images.githubusercontent.com/5219726/105863844-98306e80-5ff1-11eb-984e-d27af213c159.png)
